### PR TITLE
Fixes #9 Connection established to ZooKeeper

### LINKED ISSUE #9: Connect to zookeeper
In current helm chart we are having default zookeeper up and running. But the connector-main is not having the necessary dependency in order to connect to zookeeper. I think @ayush0x00 while implementing the zookeeper this issue will be automatically resolved.

### PR DESCRIPTION
**Pull request by ayush0x00**
This PR resolves #9. The SpringBoot application is now able to talk to zookeeper service. A Proof Of Work has been added.
Build and Run the project, then you should be able to get the instance id by visiting `http://localhost:8080/instances/app1`

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,14 +1,20 @@
 plugins {
-	id 'org.springframework.boot' version '2.3.4.RELEASE'
+	id "org.springframework.boot" version "2.6.4"
```
Comment (danishjamal104) [inline build.gradle:2]: why plugin update?

File: build.gradle [hunk 2]
```diff
@@ -28,6 +35,12 @@ dependencies {
 	implementation group: 'javax.persistence', name: 'javax.persistence-api', version: '2.2'
 }
 
+dependencyManagement {
```
Comment (danishjamal104) [inline build.gradle:38]: what this block does?

File: src/main/java/com/urlshortner/discovery/autoconfig/SpringCloudDiscoveryAutoConfig.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package com.urlshortner.discovery.autoconfig;
```
Comment (danishjamal104) [inline src/main/java/com/urlshortner/discovery/autoconfig/SpringCloudDiscoveryAutoConfig.java:1]: can we mention this configuration directly in `application.yaml`?

File: src/main/resources/META-INF/spring.factories [hunk 4]
```diff
@@ -0,0 +1,2 @@
+org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
```
Comment (danishjamal104) [inline src/main/resources/META-INF/spring.factories:1]: I think this is local files and can be added in `.gitignore`.

File: src/main/java/com/urlshortner/discovery/ServiceInstanceMapper.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package com.urlshortner.discovery;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import org.apache.curator.x.discovery.ServiceInstance;
+import org.springframework.cloud.zookeeper.discovery.ZookeeperInstance;
+
+
```
Comment (danishjamal104) [inline src/main/java/com/urlshortner/discovery/ServiceInstanceMapper.java]: Add doctstring.

File: src/main/java/com/urlshortner/discovery/ServiceRegistryImpl.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.x.discovery.ServiceDiscovery;
+import org.apache.curator.x.discovery.ServiceDiscoveryBuilder;
+import org.apache.curator.x.discovery.ServiceInstance;
+import org.springframework.cloud.zookeeper.discovery.ZookeeperDiscoveryProperties;
+import org.springframework.cloud.zookeeper.discovery.ZookeeperInstance;
+
+import java.util.Collection;
+import java.util.Optional;
+
```
Comment (danishjamal104) [inline src/main/java/com/urlshortner/discovery/ServiceRegistryImpl.java:13]: Add docstring.

File: src/main/java/com/urlshortner/discovery/Serviceregistry.java [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package com.urlshortner.discovery;
+
+public interface Serviceregistry extends AutoCloseable{
+    ServiceInstanceMapper getInstance(String name);
```
Comment (danishjamal104) [inline src/main/java/com/urlshortner/discovery/Serviceregistry.java:5]: add doc-string

**Review by danishjamal104 — CHANGES_REQUESTED** [review]
Add doctoring at the required places.

File: build.gradle (hunk 1 above)
Comment (ayush0x00) [reply build.gradle:2]: Bean was giving error with the previous version

File: build.gradle (hunk 2 above)
Comment (ayush0x00) [reply build.gradle:38]: Manages dependency for spring-cloud...will probably work without it also. But it simplifies any dependency issues in spring-cloud-zookeeper

File: src/main/java/com/urlshortner/discovery/autoconfig/SpringCloudDiscoveryAutoConfig.java (hunk 3 above)
Comment (ayush0x00) [reply src/main/java/com/urlshortner/discovery/autoconfig/SpringCloudDiscoveryAutoConfig.java:1]: Not sure....🤷‍♀️

File: src/main/resources/META-INF/spring.factories (hunk 4 above)
Comment (ayush0x00) [reply src/main/resources/META-INF/spring.factories:1]: You'll need this file ... so if we are adding it to `.gitignore` then we have to give clear Instruction for adding this file, in our README.

File: build.gradle (hunk 2 above)
Comment (danishjamal104) [reply build.gradle:38]: okay

File: src/main/resources/META-INF/spring.factories (hunk 4 above)
Comment (danishjamal104) [reply src/main/resources/META-INF/spring.factories:1]: then leave it

**Pull request merged**